internal/integration/rules.go

```diff
@@ -209,38 +209,47 @@ func addAttributes(targetMetrics *TargetMetrics, rules []AddAttributesRule) {
 type ignoreRules []IgnoreRule
 
 func (rules ignoreRules) shouldIgnore(name string, metricType metricType) bool {
-	var prefixesLen, metricTypesLen, exceptRulesLen int
+	if rules.isMetricExcludedFromFilters(name) {
```

Comment:
perhaps add a comment here explaining why the except rules are executed before the rest of the rules.